onboarding_auth/onboardingAuthData/src/main/java/com/alexafit/onboardingauthdata/repository/OnboardingAuthRepositoryImpl.kt

```diff
@@ -21,7 +26,22 @@ class OnboardingAuthRepositoryImpl(
                 handleErrorResponse(response = registerResponse)
             }
         } catch (e: Exception) {
-            Result.failure(Throwable(message = "Error in api call"))
+            Result.failure(Throwable(message = API_ERROR))
+        }
+    }
+
+    override suspend fun checkAuthentication(): Flow<Result<Unit>> {
+        val authenticationResponse = taskyApi.checkAuthentication(
+            ("Bearer " + (dataStore.authorizationKey.firstOrNull() ?: ""))
+        )
```

Comment:
Wouldn't you have to add the "Bearer" on all your API requests?

It'd be better if you just have an interceptor that adds it automatically on each api call